sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ToStringBase.scala

```diff
@@ -100,7 +100,7 @@ trait ToStringBase { self: UnaryExpression with TimeZoneAwareExpression =>
           builder.append(keyToUTF8String(keyArray.get(0, kt)).asInstanceOf[UTF8String])
           builder.append(" ->")
           if (valueArray.isNullAt(0)) {
-            if (nullString.nonEmpty) builder.append(nullString)
+            if (nullString.nonEmpty) builder.append(" " + nullString)
```

Comment:
does the codegen have the same bug?

Reply:
Yes, it does. Proposed fix here: https://github.com/apache/spark/pull/42433